### ISSUE TITLE
Dashboard empty and certain menu items giving back errors

### ISSUE THREAD

**Issue by Hydranet** [pi-hole/docker-pi-hole#1068]
This is a:  Run Issue(interface not reacting correctly)

## Details

It has been a while since I logged into the pihole web-interface but it wasn't like this last time. The dashboard is empty and not being updated anymore and when I go to certain locations in the interface I get an error back. For example when I go to the following locations I get an error
- Query Log:
```
An unknown error occurred while loading the data.
<br />
<b>Fatal error</b>:  Uncaught Error: Call to undefined function json_encode() in /var/www/html/admin/api.php:202
Stack trace:
#0 {main}
  thrown in <b>/var/www/html/admin/api.php</b> on line <b>202</b><br />
```
- When I got to local dns(dns records and cnames) I get the following error.
`DataTables warning: table id=customDNSTable - Invalid JSON response. For more information about this error, please see http://datatables.net/tn/1`
- Blacklist/Whitelist are empty while I have values there.
- Longterm data - Query Log
```
An unknown error occurred while loading the data.
<br />
<b>Warning</b>:  Use of undefined constant SQLITE3_OPEN_READONLY - assumed 'SQLITE3_OPEN_READONLY' (this will throw an Error in a future version of PHP) in <b>/var/www/html/admin/scripts/pi-hole/php/database.php</b> on line <b>61</b><br />
<br />
<b>Fatal error</b>:  Uncaught Error: Class 'SQLite3' not found in /var/www/html/admin/scripts/pi-hole/php/database.php:42
Stack trace:
#0 /var/www/html/admin/scripts/pi-hole/php/database.php(65): SQLite3_connect_try('/etc/pihole/pih...', 'SQLITE3_OPEN_RE...', true)
#1 /var/www/html/admin/api_db.php(26): SQLite3_connect('/etc/pihole/pih...')
#2 {main}
  thrown in <b>/var/www/html/admin/scripts/pi-hole/php/database.php</b> on line <b>42</b><br />

Check the server's log files (/var/log/lighttpd/error.log) for details.

You may need to increase PHP memory limit.

You can find more info in pi-hole's FAQ:
https://docs.pi-hole.net/main/faq/#error-while-loading-data-from-the-long-term-database

```
## Related Issues
- [ ] I have searched this repository/Pi-hole forums for existing issues and pull requests that look similar 

## How to reproduce the issue 

1. Environment data
  * Operating System: **Rocky Linux 8** 
  * Hardware: PC 
  * Kernel Architecture: amd64 
  * Docker Install Info and version: 
    - Software source: podman version 3.4.2 
    - Supplimentary Software: None
  * Hardware architecture: amd64 

2. docker-compose.yml contents, docker run shell command, or paste a screenshot of any UI based configuration of containers here
```
podman run -d \
 --name pihole \
 --label io.containers.autoupdate=registry \
 -e TZ="Europe/Amsterdam" \
 -e VIRTUAL_HOST="pihole.mylan.com" \
 -e PROXY_LOCATION="/admin" \
 -e ServerIP="10.31.251.102" \
 -e IPv6=false \
 -e DNSSEC=false \
 -p 53:53/tcp -p 53:53/udp \
 -p 8181:80 \
 --hostname pihole \
 --dns=127.0.0.1 --dns=1.1.1.1 --dns=9.9.9.9 \
 -v "$HOME/podman/pihole/etc-pihole/:/etc/pihole:z" \
 -v "$HOME/podman/pihole/etc-php/php:/etc/php:z" \
 -v "$HOME/podman/pihole/etc-dnsmasq.d/:/etc/dnsmasq.d:z" \
  docker.io/pihole/pihole:latest
```
- I have memory_limit set to -1 in php.ini within the container.
- I'm running it as a rootless container.

3. any additional info to help reproduce

## These common fixes didn't work for my issue

- [ V] I have tried removing/destroying my container, and re-creating a new container
- [ X] I have tried fresh volume data by backing up and moving/removing the old volume data
- [X ] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [ V] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [X ] I have tried running without my volume data mounts to eliminate volumes as the cause
When I switched to a version with an earlier tag(2022.02.1) everythings works as it did before
and when I switch to latest or the 2022.04 tag my interface is broken as described above. 
So the issue seems to be in the latest released image.

If the above debugging / fixes revealed any new information note it here.
Add any other debugging steps you've taken or theories on root cause that may help.

 Your debug token is: https://tricorder.pi-hole.net/ehKFnyUs/

**Comment by rdwebdesign** [comment]
> `Uncaught Error: Call to undefined function json_encode()`

Apparently your PHP install doesn't have JSON support. Do you have `php-json` package installed?

If not, you can install it using:

```sh
sudo apt-get install php-json
```

**Comment by PromoFaux** [comment]
Wait..

> you can install it using:

This should already be in the container, no?

**Comment by rdwebdesign** [comment]
He is using a volume pointing to `/etc/php`:

```sh
-v "$HOME/podman/pihole/etc-php/php:/etc/php:z" \
```

**Comment by Hydranet** [comment]
> > `Uncaught Error: Call to undefined function json_encode()`
> 
> Apparently your PHP install doesn't have JSON support. Do you have `php-json` package installed?
> 
> If not, you can install it using:
> 
> ```shell
> sudo apt-get install php-json
> ```
This should already be in the container image?

**Comment by rdwebdesign** [comment]
It is, but you are using your own PHP config and ignoring what we have in the image.

Remove `-v "$HOME/podman/pihole/etc-php/php:/etc/php:z"` from your config and try again.

**Comment by Hydranet** [comment]
> It is, but you are using your own PHP config and ignoring what we have in the image.
> 
> Remove `-v "$HOME/podman/pihole/etc-php/php:/etc/php:z"` from your config and try again.
I'm using that to keep the changed php memory_limit consistent because at one point pihole was crashing because of not being enough memory for php, changing that from the default to -1 fixed that.
```
root@pihole:/etc# find /etc/php -name php.ini
/etc/php/7.3/cli/php.ini
/etc/php/7.3/cgi/php.ini
```
That's why I was using that, how come image with tag 2022.04  is causing problems and the older image with the tag 2022.02.1 is working as expected? I'll try removing it to see what happens.

**Comment by rdwebdesign** [comment]
Something in your config is removing or changing JSON module config.

If you want to use your own `php.ini` file, you need to be sure the JSON module is correctly configured in it.

**Comment by Hydranet** [comment]
I had it configured like that because in the past I ran into a memory limit in php.ini  causing some problem in the pihole-FTL interface and I was told I should change the default value so I used the -v option to save my  customized php.ini version. I removed it and now it is working, will see if I run into that same issue again. The problem seems to be that in my saved config it was still using php 7.3 and in the new image 7.4 is used so there must be some in compatibility between php 7.3 and 7.4 because the only value in my php config I changed was memory_limi. Thanks for your help, I'll mark the issue as solved.
